python/lsst/obs/metadata/metadata.py

```diff
@@ -0,0 +1,179 @@
+# This file is part of obs_metadata.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Classes and support code for metadata translation"""
+
+from abc import abstractmethod
+import logging
+
+log = logging.getLogger(__name__)
+
+
+class MetadataMeta(type):
+    """Register all subclasses with the base class"""
+
+    @staticmethod
+    def _makeConstMapping(standardKey, constant):
+        def translator(header):
+            return constant
+        translator.__doc__ = f"""Returns constant value for '{standardKey}' property
+
+        Parameters
+        ----------
+        header : `dict`-like
+            Header values representing a FITS header.
+
+        Returns
+        -------
+        translation : `str` or `numbers.Number`
+            Always returns {constant}.
+        """
+        return translator
+
+    @staticmethod
+    def _makeUnitMapping(standardKey, fitsKey):
```

Comment:
I'd probably call this a "trivial" mapping rather than a "unit" mapping (and elsewhere, of course), especially because we're going to presumably use "units" to mean something like "physical units" in lots of nearby code.